### ISSUE TITLE
mountpoint is wrong

### ISSUE THREAD

**Issue by ouyang7923** [oshi/oshi#2211]
partition sda7 mountpoint result not matched.
1. partition info get by oshi:  
```
hwPartition {  
    "identification": "/dev/sda7",  
    "name": "/sys/devices/pci0000:00/0000:00:10.0/host32/target32:0:0/32:0:0:0/block/sda/sda7",  
    "type": "ext4",  
    "uuid": "7fb6f0fd-6018-406a-ae42-9c395ba6011c",  
    "size": 17714642944,  
    "major": 8,  
    "minor": 7,  
    "mountPoint": "/var"  
}  
```
2. lsblk shell command result:
```
MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT  
sda      8:0    0   65G  0 disk   
├─sda1   8:1    0  1.5G  0 part /boot  
├─sda2   8:2    0    1K  0 part   
├─sda3   8:3    0   11G  0 part /recovery  
├─sda4   8:4    0    6G  0 part [SWAP]  
├─sda5   8:5    0   15G  0 part /  
├─sda6   8:6    0   15G  0 part   
└─sda7   8:7    0 16.5G  0 part /data  
sr0     11:0    1 1024M  0 rom  
````

**Comment by dbwiddis** [comment]
OSHI gets its mount point information from `/proc/self/mounts`:
> This file lists all the filesystems currently mounted in the process's mount namespace

Further documentation indicates this is a real-time kernel view and should be accurate.

OSHI iterates and adds these to a map, so if there are multiple mountpoints for a partition, it will pick the last one in that list.  Various internet searches indicate this pseudofile is in mount order, so the "last one" should be the most recently mounted one.

`lsblk` obtains its information from sysfs via udev:
> The lsblk command reads the sysfs filesystem and udev db to gather information.

and
> Note that lsblk might be executed in time when udev does not have all information about recently added or modified devices yet. In this case it is recommended to use udevadm settle before lsblk to synchronize with udev.

Also if there are multiple mount points:
> The relationship between block devices and filesystems is not always one-to-one. The filesystem may use more block devices, or the same filesystem may be accessible by more paths. This is the reason why lsblk provides MOUNTPOINT and MOUNTPOINTS (pl.) columns. The column MOUNTPOINT displays only one mount point (usually the last mounted instance of the filesystem)

Bottom line: OSHI is faithfully reporting the mountpoint in `/proc/self/mounts`.  If that's different than `lsblk` then perhaps you can tell me a better preference to map than "the last one".

**Comment by dbwiddis** [comment]
Hey @ouyang7923 is this still an issue?

**Comment by dbwiddis** [comment]
Hi @ouyang7923, I'm going to close this issue:
1. I've identified the source of OSHI's information
2. I believe OSHI is reporting the best possible information
3. I have indicated a possible reason why there is a difference.

Please feel free to re-open this issue with a comment if you still think there's something OSHI should change.
